### PR TITLE
Updated install and production getting started docs for Bare Metal

### PR DESCRIPTION
**Pull request by chrisnegus**
*Description of changes:*
Updated the [Install EKS Anywhere](https://anywhere.eks.amazonaws.com/docs/getting-started/install/) and [Create Production Cluster](https://anywhere.eks.amazonaws.com/docs/getting-started/production-environment/) getting started guide to add Bare Metal support.

Supporting docs for this guide are being added in PRs [#2365](https://github.com/aws/eks-anywhere/pull/2365), [#2400](https://github.com/aws/eks-anywhere/pull/2400), and [#2410](https://github.com/aws/eks-anywhere/pull/2400).

I'll create a separate PR to add Tinkerbell ports to the [Ports and protocols](https://anywhere.eks.amazonaws.com/docs/reference/ports/) guide.

### CODE REVIEW

File: docs/content/en/docs/getting-started/production-environment/_index.md [hunk 1]
```diff
@@ -3,69 +3,94 @@ title: "Create production cluster"
...
+1. Set environment variables for your provider. For vSphere, these include the cluster name, and your vSphere user name and password (use single quotes around the values):
+   
+   ```bash
+   export CLUSTER_NAME=mgmt
+   export EKSA_VSPHERE_USERNAME='billy'
+   export EKSA_VSPHERE_PASSWORD='t0p$ecret'
+   ```
+   For Bare Metal, set the cluster name, IP address of the Tinkerbell stack, and the path to the hardware inventory (for example, `hardware.csv`):
+   ```bash
+   export PROVIDER=tinkerbell
```
Comment (abhinavmpandey08) [inline docs/content/en/docs/getting-started/production-environment/_index.md:67]: where does this get used?

Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/_index.md]: Removed when I split apart the getting started into two: vSphere and Bare Metal.

File: docs/content/en/docs/getting-started/production-environment/_index.md [hunk 2]
```diff
@@ -3,69 +3,94 @@ title: "Create production cluster"
...
+   export CLUSTER_NAME=mgmt
+   export CSV_FILE=<File path to hardware.csv inventory file>
+   export TINKERBELL_IP=<IP address of tinkerbell stack>
+   ```
+1. Generate a hardware configuration file (Bare Metal only), using the hardware inventory CSV file created from [Bare Metal preparation]({{< relref "/docs/reference/baremetal/bare-preparation.md" >}}):
+   ```bash
+   eksctl anywhere generate hardware --filename $CSV_FILE --tinkerbell-ip $TINKERBELL_IP
+   ```
+   The command above will place a `hardware.yaml` file in your working directory. This file will be used as an input when you generate the cluster config.
```
[the remark this answers is not in the document]
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/_index.md]: Removed when I split the getting started into two files.

File: docs/content/en/docs/getting-started/install/_index.md [hunk 3]
```diff
@@ -4,8 +4,7 @@ weight: 10
 ---
 
 EKS Anywhere will create and manage Kubernetes clusters on multiple providers.
-Currently we support creating development clusters locally with Docker and production clusters using VMware vSphere.
-Other deployment targets will be added in the future, including bare metal support in 2022.
+Currently we support creating development clusters locally with Docker and production clusters using Bare Metal or VMware vSphere.
```
Comment (abhinavmpandey08) [inline docs/content/en/docs/getting-started/install/_index.md:7]: nit
either
```suggestion
Currently we support creating development clusters locally with Docker and production clusters with Bare Metal or VMware vSphere.
```
or
```suggestion
Currently we support creating development clusters locally using Docker and production clusters using Bare Metal or VMware vSphere.
```

File: docs/content/en/docs/getting-started/production-environment/baremetal-getstarted.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
+---
+title: "Create Bare Metal production cluster"
+weight: 40
+---
+
+EKS Anywhere supports a Bare Metal provider for production grade EKS Anywhere deployments.
+EKS Anywhere allows you to provision and manage Amazon EKS on your own infrastructure.
```
Comment (abhinavmpandey08) [inline docs/content/en/docs/getting-started/production-environment/baremetal-getstarted.md:7]: > manage Amazon EKS on your own infrastructure

To me this sounds like, you would able to manage EKS on the cloud from your own infra using EKS-A.

I would re-phrase it to maybe say something along these lines
```suggestion
EKS Anywhere allows you to provision and manage Amazon EKS like Kubernetes environment in your own infrastructure.
```

File: docs/content/en/docs/getting-started/install/_index.md (hunk 3 above)
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/install/_index.md:7]: Done. I used the second suggestion.

File: docs/content/en/docs/getting-started/production-environment/baremetal-getstarted.md (hunk 4 above)
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/baremetal-getstarted.md:7]: How about this: "EKS Anywhere allows you to provision and manage Kubernetes clusters based on Amazon EKS software on your own infrastructure."
Feel free to make other wording suggestions.

File: docs/content/en/docs/getting-started/production-environment/baremetal-getstarted.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+This document walks you through setting up EKS Anywhere as a self-managed cluster.
+It does not yet support the concept of a separate management cluster for managing one or more workload clusters.
+
+
+## Prerequisite checklist
+
+EKS Anywhere needs:
+
+* To be run on an Admin machine that has certain [machine requirements]({{< relref "../install" >}}).
```
[the remark this answers is not in the document]
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/baremetal-getstarted.md:17]: Great point! It's in a few places, but it should be in the install doc that link points to. I added this sentence to the "Administrative machine prerequisites" section: "- Administrative machine must be on the same Layer 2 network as the cluster machines (Bare Metal provider only)."

**Review by abhinavmpandey08 — APPROVED** [review]
/lgtm

**Comment by vipin-mohan** [comment]
Notes:
- Can we change the title to "Create a production cluster on Bare Metal" (similarly "Create a production cluster on VMware vSphere")?
- Change "This document walks you through setting up EKS Anywhere as a self-managed cluster. It does not yet support the concept of a separate management cluster for managing one or more workload clusters." to "This document walks you through setting up EKS Anywhere on bare metal."
- There are two instances where we say "--kube-version 1.21". Is there any reason this can't be 1.22? I'd prefer we go with the latest Kubernetes version (unless there's a technical issue that I'm not aware of)

**Pull request merged**